sdk/core/System.ClientModel/src/Experimental/PipelineProtocolExtensions.cs

```diff
@@ -42,51 +42,4 @@ public static PipelineResponse ProcessMessage(this ClientPipeline pipeline, Pipe
 
         throw new ClientRequestException(message.Response);
     }
-
-    public static async ValueTask<NullableOutputMessage<bool>> ProcessHeadAsBoolMessageAsync(this ClientPipeline pipeline, PipelineMessage message, RequestOptions requestContext)
```

Comment:
If we decide to take this change to avoid shipping an [EBN] property in a System namespace, we will need to establish a new ClientModel client convention for the return type from HEAD requests, which today in Azure.Core-based clients return `NullableResponse<bool>`